.gitlab/image_deploy/docker_linux.yml

```diff
@@ -181,86 +143,64 @@ dev_master-a7:
         IMG_DESTINATIONS: agent-dev:master-py3-jmx
 
 dev_master-dogstatsd:
-  extends: .docker_job_definition
+  extends: .docker_publish_job_definition
   stage: image_deploy
   rules:
     !reference [.on_main_a7]
   needs:
     - docker_build_dogstatsd_amd64
-  trigger:
```

Comment:
I see you are replacing all uses of the `trigger` gitlab keyword with regular jobs that call `inv pipeline.trigger-child-pipeline` (as defined in `docker_publish_job_definition`).

This will still work, but it has the following implications:
- The child pipelines might take longer to get triggered if all workers are busy (since these jobs now need a runner to trigger the child pipeline from bash)
- If the child pipeline fails, the job on the main pipeline will still appear green (because it succeeded triggering the child pipeline, even if the child pipeline failed). This is something we could/should fix inside `trigger-child-pipeline`, by waiting for the child pipeline to complete and bubbling up the status, but until we do that it might hide errors.

Reply:
The initial motivation for using `inv pipeline.trigger-child-pipeline` instead of the `trigger` GitLab keyword was that the `trigger` keyword doesn’t allow complex logic for setting environment variables.
With this change, pretty much all jobs need the logic of suffixing the docker bucket with `-release` if the build is a build of a tagged commit.
So, this logic has been factorised in [`.docker_publish_job_definition`](https://github.com/DataDog/datadog-agent/pull/8560/files#diff-239bf1835b577f11028b131aadce4b3b609380c7ebcc9f01d3ddef3dcac93212R14).